operators/pkg/controller/elasticsearch/driver/downscale.go

```diff
@@ -249,17 +249,42 @@ func updateZenSettingsForDownscale(ctx downscaleContext, downscale ssetDownscale
 		return nil
 	}
 
-	// TODO: only update in case 2->1 masters.
-	// Update Zen1 minimum master nodes API, accounting for the updated downscaled replicas.
-	_, err := zen1.UpdateMinimumMasterNodes(ctx.k8sClient, ctx.es, ctx.esClient, actualStatefulSets, ctx.reconcileState)
-	if err != nil {
+	// Maybe update zen1 minimum_master_nodes.
+	if err := maybeUpdateZen1ForDownscale(ctx, actualStatefulSets); err != nil {
 		return err
 	}
 
-	// Update zen2 settings to exclude leaving master nodes from voting.
+	// Maybe update zen2 settings to exclude leaving master nodes from voting.
 	if err := zen2.AddToVotingConfigExclusions(ctx.esClient, downscale.statefulSet, downscale.leavingNodeNames()); err != nil {
 		return err
 	}
 
 	return nil
 }
+
+// maybeUpdateZen1ForDownscale updates zen1 minimum master nodes if we are downscaling from 2 to 1 master node.
+func maybeUpdateZen1ForDownscale(ctx downscaleContext, actualStatefulSets sset.StatefulSetList) error {
+	if !zen1.AtLeastOneNodeCompatibleWithZen1(actualStatefulSets) {
+		return nil
+	}
+	// If we are moving from 2 to 1 master nodes, we need to update minimum_master_nodes before removing
+	// the 2nd node, otherwise the cluster won't be able to form anymore.
+	// This is inherently unsafe (can cause split brains), but there's no alternative.
```

Comment:
Few questions here:
Do I understand correctly that this can just happen and we can't do anything about it? 
Is this the same possible for non-ECK ES?
What is the user experience like? After all, they would be making a valid change and end up with broken cluster?
Can we detect this? Communicate to user somehow?
Should we not allow this if it's not safe?

Reply:
> Do I understand correctly that this can just happen and we can't do anything about it?
Is this the same possible for non-ECK ES?

Yes. This is one of the flaw of zen1. Hurray zen2! 😄 

> What is the user experience like? 

ES users are not advised to create ES clusters with 2 master nodes.

> Should we not allow this if it's not safe?

Technically, we could. I don't think we want to take this responsibility though, this is ES responsibility IMO. In general we should make it so we can run ES as natively as possible, with its bugs and flaws, and not try to workaround them ourselves too much.

> Can we detect this? Communicate to user somehow?

Maybe we could emit an event here, that's a good point 👍

Reply:
Ok, got it, thanks.

> In general we should make it so we can run ES as natively as possible, with its bugs and flaws

+1, the event could mention the operation itself being risky.